src/ipc/ipc-common/LocalProviderBinder.h

```diff
@@ -66,9 +66,9 @@ class LocalProviderBinder : public QObject
     void onLocalAdapterUnavailable(QString objectPath, NewIPCServiceAdapterBase *adapter)
     {
         Q_UNUSED(objectPath);
-        auto service = m_interfaceManager.serviceMatches(m_proxy.objectPath(), adapter);
-        if (service) {
+        if (m_adapter == adapter) {
```

Comment:
Please add comment telling that you compare pointers, not values. Now it is clear, but after some time may look like error.